src/core/time.d

```diff
@@ -611,7 +622,7 @@ public:
                   subtract this $(D Duration) from.
       +/
     Duration opBinaryRight(string op, D)(D lhs) const nothrow @nogc
-        if((op == "+" || op == "-") &&
+        if((op == "+" || op == "-" || op == "%") &&
             is(_Unqual!D == TickDuration))
     {
         return Duration(mixin("lhs.hnsecs " ~ op ~ " _hnsecs"));
```

Comment:
Considering that we just talked about how `typeof(a % b) == typeof(a)`, I'm thinking this support for `TickDuration % Duration -> Duration` is ill advised anyway. I think it will not be a bad thing to see `TickDuration` and this support go.